front/app/containers/Admin/reporting/containers/ReportLanguageProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+
+// Utils
+import { IntlProvider } from 'react-intl';
+import { isNilOrError, NilOrError } from 'utils/helperUtils';
+import moment from 'moment';
+
+// Typings
+import { Locale } from 'typings';
+
+type props = {
+  reportLocale: Locale | NilOrError;
+  platformLocale: Locale | NilOrError;
+  children: React.ReactNode;
+};
+
+// Loads correct messages for the report locale & change moment locale
+const ReportLanguageProvider = ({
+  reportLocale,
+  platformLocale,
+  children,
+}: props) => {
+  const [messages, setMessages] = useState();
+
+  useEffect(() => {
+    if (!isNilOrError(reportLocale)) {
+      // Note: seems inefficient when we have already loaded all these in the main LanguageProvider
+      import(`i18n/${reportLocale}`).then((translationMessages) => {
+        setMessages(translationMessages.default);
+      });
+      moment.locale(reportLocale);
+    }
+
+    return () => {
+      // Set the moment locale back to the platform locale on cleanup
+      if (!isNilOrError(platformLocale)) {
+        moment.locale(platformLocale);
```

Comment:
Ok, but this actually completely overrides the global moment locale until the whole report unmounts... in theory it should be fine since I don't think we use any locale-specific date formatting in the full-screen editor (except for the report itself, which is why we need this)... but still, a bit of a dirty solution that could cause very strange bugs. I also wouldn't put the `isNilOrError(platformLocale)` check in here- if it's `undefined` for whatever reason the moment global locale doesn't get reset and it gets out of sync with the rest of the platform)... not sure if that can even happen but I don't like that we have to worry about this kind of stuff